internal/cli/cli.go

```diff
@@ -84,6 +85,10 @@ func RunPreflight(
 
 	logger.Info(fmt.Sprintf("Preflight result: %s", convertPassedOverall(results.PassedOverall)))
 
+	if !results.PassedOverall && viper.Instance().GetBool("exit_with_failure") {
+		return fmt.Errorf("one or more checks did not pass")
```

Comment:
What about this... How about we just return an error with results.PassedOverall and leave the viper out of this.

Check the error and the bool further up without introducing viper here.

Reply:
I think the intent here is so that we don't break peoples existing CI that depend on our error codes, and we make this an opt in feature via a flag.

Reply:
I'm not suggesting we get rid of the flag... I'm suggesting we don't push the flag down into the cli package. 

For instance... The cli package would now have a `cli.DidNotPassOverallError`. (Please don't use that name)

```
if !results.PassedOverall {
  return cli.DidNotPassOverallError
}
```

_Meanwhile, somewhere in the cmd tree_

```
if errors.Is(err, cli.DidNotPassOverallError) && viper.Instance().GetBool("exit_with_failure") {
  return err
}

// Check for other errors as usual
```

This keeps the existence of this flag from escaping cmd.